### PR TITLE
rf: less dom queries using bind:this

### PR DESCRIPTION
**Pull request by Carlos-err406**
## Pull Request Checklist

- [x] **Description:** Briefly describe the changes in this pull request.
- [ ] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [ ] **Documentation:** Have you updated relevant documentation?
- [ ] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?

---

## Description

addresses issue #1020 about querying the DOM

refactored the dom queries (document.getElementById) for a more svelte-ish approach using bind:this

---

### Added

-

### Fixed

- centralized the way to access the dom elements in a more efficient and clean manner

### Changed

- the way to query the dom

### Removed

- some of the manual dom queries accross the codebase

### CODE REVIEW

File: src/lib/components/chat/MessageInput.svelte [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <script lang="ts">
 	import { toast } from 'svelte-sonner';
 	import { onMount, tick } from 'svelte';
-	import { settings } from '$lib/stores';
```
Comment (tjbck) [inline src/lib/components/chat/MessageInput.svelte:4]: I'd suggest not to use stores and to keep variables for dom elements in the components.

Comment (Carlos-err406) [reply src/lib/components/chat/MessageInput.svelte:4]: what if the element is queried from outside the component?

**Comment by tjbck** [comment]
Creating stores for each elements only adds additional code without any real tangible benefits, I suggest we only modify the areas that could be addressed with private variables within the component.

File: src/lib/components/chat/MessageInput.svelte (hunk 1 above)
Comment (tjbck) [reply src/lib/components/chat/MessageInput.svelte:4]: I see your point of possible performance gains but we should keep the code as-is.

Comment (Carlos-err406) [reply src/lib/components/chat/MessageInput.svelte:4]: roger that

**Comment by Carlos-err406** [comment]
> Creating stores for each elements only adds additional code without any real tangible benefits, I suggest we only modify the areas that could be addressed with private variables within the component.

in the case of chatTextArea (wich is like half the dom queries) i think it actually reduces the possible inconsistencies

but, maybe is not a good time for major changes in the code, i'll rollback the store ones then

File: src/lib/components/chat/Messages/UserMessage.svelte [hunk 2]
```diff
@@ -20,18 +20,17 @@
 
 	let edit = false;
 	let editedContent = '';
-
+	let messageEditTextArea: HTMLTextAreaElement;
```
Comment (tjbck) [inline src/lib/components/chat/Messages/UserMessage.svelte:23]: We might want to name it `messageEditTextAreaElement` to denote that it's an element variable!

Comment (Carlos-err406) [reply src/lib/components/chat/Messages/UserMessage.svelte:23]: i missed that one, fixing now

File: src/lib/components/chat/MessageInput.svelte [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
 	export let suggestionPrompts = [];
 	export let autoScroll = true;
-
+	let chatTextArea:HTMLTextAreaElement
```
Comment (tjbck) [inline src/lib/components/chat/MessageInput.svelte:22]: Another one!

File: src/routes/(app)/+layout.svelte [hunk 4]
```diff
@@ -37,7 +37,7 @@
 
 	let ollamaVersion = '';
 	let loaded = false;
-
+	let showShortcutsButton:HTMLButtonElement
```
Comment (tjbck) [inline src/routes/(app)/+layout.svelte:40]: +1

File: src/routes/(app)/c/[id]/+page.svelte [hunk 5]
```diff
@@ -45,7 +45,7 @@
 	let stopResponseFlag = false;
 	let autoScroll = true;
 	let processing = '';
-
+	let messagesContainerELement: HTMLDivElement;
```
Comment (tjbck) [inline src/routes/(app)/c/[id]/+page.svelte:48]: Typo

Comment (Carlos-err406) [reply src/routes/(app)/c/[id]/+page.svelte]: fixed

File: src/routes/(app)/+layout.svelte (hunk 4 above)
Comment (Carlos-err406) [reply src/routes/(app)/+layout.svelte]: fixed

File: src/lib/components/chat/MessageInput.svelte (hunk 3 above)
Comment (Carlos-err406) [reply src/lib/components/chat/MessageInput.svelte]: fixed

**Comment by tjbck** [comment]
LGTM, Thanks!

**Comment by justinh-rahb** [comment]
Due to the number of related issues, it’s plainly obvious there was no thorough testing on this PR. We value efficiency over aesthetics, ugly vanilla JS that works will always be better than fancy JS that doesn't. Going forward, please ensure that all code changes you introduce have actually been tested as it's clear that things were overlooked here.

**Pull request merged**